src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs

```diff
@@ -39,5 +39,19 @@ public async Task<FormulaValue> EvalAsync(RecordValue parameters, CancellationTo
                 return maxCallDepthException.ToErrorValue(_irnode.IRContext);
             }
         }
+
+        public async Task<FormulaValue> EvalAsync(RecordValue parameters, CancellationToken cancel, StackDepthCounter stackMarker)
+        {
+            var ev2 = new EvalVisitor(_cultureInfo, cancel);
+            try
+            {
+                var newValue = await _irnode.Accept(ev2, new EvalVisitorContext(SymbolContext.NewTopScope(_topScopeSymbol, parameters), stackMarker));
+                return newValue;
+            }
+            catch (MaxCallDepthException maxCallDepthException)

```

Comment:
This call is for evaluating a UDF, right? Should the exception be caught, or should it be thrown up the whole call stack to the top `EvalAsync`

Reply:
+1
Rename this overload to EvalInternalAsync
Remove catch here, keep the one on line 37. 

Add comment :)

Reply:
Fixed